### PR TITLE
Added poetry for setup and workflow

### PR DESCRIPTION
**Pull request by mwbrulhardt**

### CODE REVIEW

**Comment by mwbrulhardt** [comment]
Hey it looks like the PR was removed for some reason. But sounds good, I made it solely poetry based and also added that to the workflow as well so it would be compatible. I also bumped the version to `0.0.4`. Let me know if you need me to do anything else. Thanks for taking the time to review as well!

File: pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,36 @@
...
+name = "pylambertw"
+version = "0.0.4"
+description = "Python implementation of the Lambert W x F framework for analyzing skewed, heavy-tailed distribution with an sklearn interface and torch based maximum likelihood estimation (MLE)."
+authors = ["Georg M. Goerg <im@gmge.org>"]
+homepage = "https://github.com/gmgeorg/pylambertw.git"
+
+[tool.poetry.dependencies]
+python = "^3.9"
+
+numpy = ">=1.0.1"
```
Comment (gmgeorg) [inline pyproject.toml:13]: nit: can you add this in alphabetical order

Comment (gmgeorg) [inline pyproject.toml:13]: see if you require the cython requirement (new requirements.txt file) .  w/o it I had trouble with scipy CI/CD tests passing; if it passes w/o cython requirement as-is, feel free to drop that requirement

**Review by gmgeorg — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (mwbrulhardt) [reply pyproject.toml:13]: Ok will do!

Comment (mwbrulhardt) [reply pyproject.toml:13]: Ok I tried it with and without that requirement and it seems the CI/CD pipeline passes in both cases. I'll leave it out for now, and if it becomes a problem later you can always add it in.

**Review by gmgeorg — APPROVED** [review]

**Pull request merged**